src/ImageSharp/Metadata/Profiles/XMP/XmpProfile.cs

```diff
@@ -0,0 +1,199 @@
+
+            if (other.Data != null)
+            {
+                this.Data = new byte[other.Data.Length];
+                other.Data.AsSpan().CopyTo(this.Data);
+            }
+        }
+
+        /// <summary>
+        /// Gets the rax XML document containing the XMP profile.
+        /// </summary>
+        public XDocument Document
+        {
+            get
+            {
+                this.InitializeDocument();
+                return this.document;
+            }
+        }
+
+        /// <summary>
+        /// Gets the byte data of the XMP profile.
+        /// </summary>
+        public byte[] Data { get; private set; }
+
+        /// <summary>
+        /// Checks whether two <see cref="XmpProfile"/> structures are equal.
+        /// </summary>
+        /// <param name="left">The left hand <see cref="XmpProfile"/> operand.</param>
+        /// <param name="right">The right hand <see cref="XmpProfile"/> operand.</param>
+        /// <returns>
+        /// True if the <paramref name="left"/> parameter is equal to the <paramref name="right"/> parameter;
+        /// otherwise, false.
+        /// </returns>
+        public static bool operator ==(XmpProfile left, XmpProfile right)
+        {
+            if (ReferenceEquals(left, right))
+            {
+                return true;
+            }
+
+            if (left is null)
+            {
+                return false;
+            }
+
+            return left.Equals(right);
+        }
+
+        /// <summary>
+        /// Checks whether two <see cref="XmpProfile"/> structures are not equal.
+        /// </summary>
+        /// <param name="left">The left hand <see cref="XmpProfile"/> operand.</param>
+        /// <param name="right">The right hand <see cref="XmpProfile"/> operand.</param>
+        /// <returns>
+        /// True if the <paramref name="left"/> parameter is not equal to the <paramref name="right"/> parameter;
+        /// otherwise, false.
+        /// </returns>
+        public static bool operator !=(XmpProfile left, XmpProfile right) => !(left == right);
+
+        /// <inheritdoc/>
+        public XmpProfile DeepClone() => new(this);
+
+        /// <summary>
+        /// Updates the data of the profile.
+        /// </summary>
+        public void UpdateData()
+        {
+            if (this.document == null)
+            {
+                return;
+            }
+
+            int initialLength = 256;
+            if (this.Data != null)
+            {
+                initialLength = this.Data.Length;
+            }
+
+            using var stream = new MemoryStream(initialLength);
+            using var writer = new StreamWriter(stream, Encoding.UTF8);
+            this.document.Save(writer);
+            this.Data = stream.ToArray();
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode() => base.GetHashCode();
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
```

Comment:
I don't see equality being implemented for other profiles. Also see previous comment. We should drop this feature IMO.

Reply:
Yeah, definitely no need for this.

Reply:
Some background on why there are equality checks in `XmpProfile`

Other profiles can rely on their byte array equality check.
In XMP however, there can be non-relevant changes, like BOM marker, whitespace, ordering of elements and attributes.

All these **_do_** change the byte array, but **_do not_** change the XML. Hence these should not be causing the equality check of `XmpProfile` to return `false`.

Reply:
My point is that other profiles do not implement equality in any form (neither byte-array, nor semantic). I don't see how / why should a complex type like this implement equality out of the box. If a user needs it, they can implement XML-based comparison in their own code.
Out of curiosity: Do you have an important personal use case here?

Reply:
Equality methods and operators are removed now.